bin/retrieve_dns.py

```diff
@@ -223,41 +245,41 @@ def runprocess(config_file, log_config_file):
     xml_string = None
     fetch_failed = False
 
-    next_url = cfg.gocdb_url
-    try:
-        # If next_url is none, it implies we have reached the end of paging
-        # (or that paging was not turned on).
-        # The addition of 'not fetch_failed' catches the case where no XML is
-        # returned from next_url (i.e. gocdb_url).
-        while next_url is not None and not fetch_failed:
-            xml_string = get_xml(next_url, cfg.proxy)
-            log.info("Fetched XML from %s", next_url)
-
-            try:
-                # Parse the XML into a Document Object Model
-                dom = xml.dom.minidom.parseString(xml_string)
-                # Get the next url, if any
-                next_url = next_link_from_dom(dom)
-                # Add the listed DNs to the list
-                dns.extend(dns_from_dom(dom))
-            except xml.parsers.expat.ExpatError, e:
-                log.warn('Failed to parse the retrieved XML.')
-                log.warn('Is the URL correct?')
-                fetch_failed = True
-    except AttributeError:
-        # gocdb_url == None
-        log.info("No GOCDB URL specified - won't fetch URLs.")
-    except IOError, e:
-        log.info("Failed to retrieve XML - is the URL correct?")
-        log.info(e)
-        fetch_failed = True
-
-    if fetch_failed and (time.time() - os.path.getmtime(cfg.dn_file) <
-                         (cfg.expire_hours * 3600)):
-        log.warn('Failed to update DNs from GOCDB. Will not modify DNs file.')
-        log.info("auth will exit.")
-        log.info(LOG_BREAK)
-        sys.exit(1)
+    for next_url in generate_gocdb_urls(cfg.gocdb_url,cfg.service_types):
```

Comment:
As discussed in the hackathon, what happens here if `cfg.gocdb_url` contains a service type because a user hasn't updated their config?